src/shared/util.js

```diff
@@ -1219,16 +1219,111 @@ function MessageHandler(sourceName, targetName, comObj) {
   this.targetName = targetName;
   this.comObj = comObj;
   this.callbackIndex = 1;
+  this.streamId = 1;
   this.postMessageTransfers = true;
-  var callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
-  var ah = this.actionHandler = Object.create(null);
+  let streamSinks = this.streamSinks = Object.create(null);
+  let streamControllers = this.streamControllers = Object.create(null);
+  let callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
+  let ah = this.actionHandler = Object.create(null);
 
   this._onComObjOnMessage = (event) => {
-    var data = event.data;
+    let data = event.data;
     if (data.targetName !== this.sourceName) {
       return;
     }
-    if (data.isReply) {
+    if (data.stream) {
+      let sourceName = this.sourceName;
+      let targetName = data.sourceName;
+      let streamId = data.streamId;
+      switch (data.stream) {
+        case 'start_complete':
+          if (data.success === true || data.success === 'true') {
```

Comment:
> Just to be at safer side, i thought maybe somehow while sending data between main and worker, success will change to string. But i am not sure.

That shouldn't be a problem, since the [structured cloning algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) shouldn't change types for something as simple as a Boolean.

If you want to check for yourself, look at the value/type of `data.isReply` in the current code.